### PR TITLE
Update python-integration-tests.yml

### PR DESCRIPTION
**Pull request by shawncal**
Adding back the paths filter on the Python integration tests GitHub actions.

This ensures that the tests only run when changes under the python/ folder are included. Otherwise, it skips to the final validation step.

### CODE REVIEW

**Review by lemillermicrosoft — APPROVED** [review]
Sweet, let's change the matrix for `python-merge-gate` to ubuntu only -- could be a different PR.

**Comment by awharrison-28** [comment]
since it's just one interation, no matrix, let's run all the tests for merge-gate. 

```
cd python
poetry run pytest ./tests/integration -v
```

**Review by awharrison-28 — APPROVED** [review]

**Review by lemillermicrosoft — APPROVED** [review]

**Review by dmytrostruk — APPROVED** [review]

File: .github/workflows/python-integration-tests.yml [hunk 1]
```diff
@@ -17,15 +17,36 @@ permissions:
...
+        if: steps.filter.outputs.python == 'true'
+        run: echo "Python file"
+      # run only if not 'python' files were changed
+      - name: not python tests
+        if: steps.filter.outputs.python != 'true'
+        run: echo "NOT python file"
+
   python-merge-gate:
-    if: ${{ github.event_name != 'pull_request' && github.event_name != 'schedule'}}
+    if: ${{ github.event_name != 'pull_request' && github.event_name != 'schedule' && needs.paths-filter.outputs.pythonChanges == 'true' }}
```
Comment (lemillermicrosoft) [inline .github/workflows/python-integration-tests.yml:42]: Feels like this didn't work quite right... Maybe I'm missing something.

https://github.com/microsoft/semantic-kernel/actions/runs/5955500956/job/16154270036

Comment (lemillermicrosoft) [reply .github/workflows/python-integration-tests.yml:42]:
```suggestion
    if: ${{ github.event_name != 'pull_request' && github.event_name != 'schedule'}} && needs.paths-filter.outputs.pythonChanges == 'true' 
```

File: .github/workflows/python-integration-tests.yml [hunk 2]
```diff
@@ -70,16 +91,10 @@ jobs:
...
-          poetry run pytest ./tests/integration/completions/test_hf_local_text_completions.py -v
-          poetry run pytest ./tests/integration/connectors/memory/test_chroma.py -v
-          poetry run pytest ./tests/integration/connectors/memory/test_qdrant_memory_store.py -v
-          poetry run pytest ./tests/integration/planning -v
-          poetry run pytest ./tests/integration/embeddings -v
+          poetry run pytest ./tests/integration -v
 
   python-integration-tests:
-    if: ${{ github.event_name == 'schedule' || github.event_name == 'workflow_dispatch' }}
+    if: ${{ (github.event_name == 'schedule' || github.event_name == 'workflow_dispatch') && needs.paths-filter.outputs.pythonChanges == 'true' }}
```
Comment (lemillermicrosoft) [inline .github/workflows/python-integration-tests.yml:97]:
```suggestion
    if: (github.event_name == 'schedule' || github.event_name == 'workflow_dispatch') && needs.paths-filter.outputs.pythonChanges == 'true' 
```

**Pull request merged**